### ISSUE TITLE
False-positive for `missing_safety_doc` with block-comments

### ISSUE THREAD

**Issue by snyball** [rust-lang/rust-clippy#5731]
With the following code:

```rust
/**
 * Drop an external reference
 *
 * # Arguments
 *
 * - `id` : Reference ID
 *
 * # Safety
 *
 * Using the reference after it has been dropped is UB.
 */
pub unsafe fn drop_ext(&mut self, id: u32) {
    self.extref.remove(&id);
}
```

I get this warning:

```rust
warning: unsafe function's docs miss `# Safety` section
   --> src/nkgc.rs:779:5
    |
779 | /     pub unsafe fn drop_ext(&mut self, id: u32) {
780 | |         self.extref.remove(&id);
781 | |     }
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc
```

This code does not result in a clippy warning:

```rust
///
/// Drop an external reference
///
/// # Arguments
///
/// - `id` : Reference ID
///
/// # Safety
///
/// Using the reference after it has been dropped is UB.
///
pub unsafe fn drop_ext(&mut self, id: u32) {
    self.extref.remove(&id);
}
```

Given that `/** ... */`-style doc-comments are supported by rustdoc, I would expect them to be supported by clippy as well.

**Comment by flip1995** [comment]
Duplicate of #5593

**Comment by flip1995** [comment]
A fix is on its way: #5659
